runner/src/runner.rs

```diff
@@ -626,4 +687,55 @@ mod tests {
         expected_output.push('\n');
         assert_eq!(output.stdout, expected_output.as_bytes());
     }
+
+    fn set_mocks(
+        runner: &mut MockRunner,
+        shutdown_flag: Arc<AtomicBool>,
+        shutdown_flag_count: usize,
+        terminate_proc_count: usize,
+    ) {
+        let tmp = TempDir::new("runner-tests").expect("failed to create tempdir");
```

Comment:
It just dawned on me that you could have some form of storage trait which would allow tests to use an in-memory store instead of actually writing to a temporary directory. My concern otherwise is contamination since there is no cleanup if these tests fail.

Reply:
> It just dawned on me that you could have some form of storage trait which would allow tests to use an in-memory store instead of actually writing to a temporary directory. My concern otherwise is contamination since there is no cleanup if these tests fail.

Not true: The [TempDir](https://docs.rs/tempdir/latest/tempdir/struct.TempDir.html) type creates a directory on the file system that is deleted once it goes out of scope.

Reply:
The [tempfile](https://docs.rs/tempfile/latest/tempfile/#design) crate docs suggest there is no better way of creating a temporary file while also knowing its location. `TempDir` uses the `/tmp` directory which is cleared on reboot.

I looked for some storage mocks, but that only led me to substrate and cosmos docs.